src/parsec.rs

```diff
@@ -43,13 +77,15 @@ impl ParsecMap {
             *gen_pfx_info.first_info.version(),
             create(full_id, gen_pfx_info),
         );
+        let size_counter = ParsecSizeCounter::default();
 
-        Self { map }
+        Self { map, size_counter }
     }
 
     pub fn init(&mut self, full_id: FullId, gen_pfx_info: &GenesisPfxInfo, log_ident: &LogIdent) {
         if let Entry::Vacant(entry) = self.map.entry(*gen_pfx_info.first_info.version()) {
             let _ = entry.insert(create(full_id, gen_pfx_info));
+            self.size_counter = ParsecSizeCounter::default();
```

Comment:
I'm not sure if any of your test valid this?

Reply:
Good point, I added a fixup commit that adds a check for this